src/stateMachines/OID4VCIMachine.tsx

```diff
@@ -0,0 +1,534 @@
+  const {selectedCredentials} = _ctx;
+  return selectedCredentials !== undefined && selectedCredentials.length > 0;
+};
+
+export const createOID4VCIMachine = (opts?: CreateOID4VCIMachineOpts): OID4VCIStateMachine => {
+  const {requestData} = opts ?? {};
+
+  // TODO this needs to be improved
+  if (!requestData) {
+    throw new Error('Unable to create OID4VCI machine. Missing request data');
+  }
+
+  const initialContext: OID4VCIMachineContext = {
+    // TODO we need to store the data from OpenIdProvider here in the context and make sure we can restart the machine with it and init the OpenIdProvider
+    requestData,
+    supportedCredentials: [],
+    selectedCredentials: [],
+    credentialOffers: [],
+    hasContactConsent: true,
+    contactAlias: '',
+  };
+
+  return createMachine<OID4VCIMachineContext, OID4VCIMachineEventTypes>({
+    id: 'OID4VCI',
+    predictableActionArguments: true,
+    initial: OID4VCIMachineStates.initiating,
+    schema: {
+      events: {} as OID4VCIMachineEventTypes,
+      guards: {} as
+        | {type: OID4VCIMachineGuards.hasNotContactGuard}
+        | {type: OID4VCIMachineGuards.selectCredentialsGuard}
+        | {type: OID4VCIMachineGuards.authenticationGuard}
+        | {type: OID4VCIMachineGuards.contactIdentityGuard}
+        | {type: OID4VCIMachineGuards.verificationCodeGuard}
+        | {type: OID4VCIMachineGuards.hasContactGuard}
+        | {type: OID4VCIMachineGuards.createContactGuard}
+        | {type: OID4VCIMachineGuards.hasSelectedCredentialsGuard},
+      services: {} as {
+        [OID4VCIMachineServices.initiating]: {
+          data: OpenId4VcIssuanceProvider;
+        };
+        [OID4VCIMachineServices.createCredentialSelection]: {
+          data: Array<ICredentialTypeSelection>;
+        };
+        [OID4VCIMachineServices.retrieveContact]: {
+          data: IContact | undefined;
+        };
+        [OID4VCIMachineServices.retrieveCredentials]: {
+          data: Array<MappedCredentialOffer> | undefined;
+        };
+        [OID4VCIMachineServices.addContactIdentity]: {
+          data: void;
+        };
+        [OID4VCIMachineServices.verifyCredentials]: {
+          data: void;
+        };
+        [OID4VCIMachineServices.storeCredentialBranding]: {
+          data: void;
+        };
+        [OID4VCIMachineServices.storeCredentials]: {
+          data: void;
+        };
+      },
+    },
+    context: {
+      ...initialContext,
+    } as OID4VCIMachineContext,
+    states: {
+      [OID4VCIMachineStates.initiating]: {
+        id: OID4VCIMachineStates.initiating,
+        invoke: {
+          src: OID4VCIMachineServices.initiating,
+          onDone: {
+            target: OID4VCIMachineStates.creatingCredentialSelection,
+            actions: assign({openId4VcIssuanceProvider: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<any>) => _event.data}),
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
```

Comment:
i don't think i am following this. this is a state to show a certain screen. just like any other screen in the machine. it is there to show any error we have cause that is how the flow (orginal flow) was designed

Reply:
The state is introduced to handle an error. In this case it is a human that sees the error and then you typically move to an end state. In a machine to machine context, the state might only log for instance or return an error to the calling party via a JSON payload. 

So the comment was not related to the state being present, it is more that the name 'show' suggests to implementation what they should do, instead of actually indicating a state the machine is in

Reply:
as discussed, changed it too handleError